File: app/src/main/java/remix/myplayer/bean/mp3/Folder.java
package remix.myplayer.bean.mp3;

/**
 * Created by Remix on 2018/1/9.
 */

public class Folder {
    private String Name;
    private int Count;
    private String Path;

    public Folder(String name, int count, String path) {
        Name = name;
        Count = count;
        Path = path;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public int getCount() {
        return Count;
    }

    public void setCount(int count) {
        Count = count;
    }

    public String getPath() {
        return Path;
    }

    public void setPath(String path) {
        Path = path;
    }
}

File: app/src/main/java/remix/myplayer/bean/mp3/Folder.kt
package remix.myplayer.bean.mp3

import kotlinx.serialization.Serializable
import java.io.Serial

/**
 * Created by Remix on 2018/1/9.
 */
@Serializable
data class Folder(val name: String?,
                  val count: Int,
                  val path: String) : APlayerModel {

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as Folder

    return path == other.path
  }

  override fun hashCode(): Int {
    return path.hashCode()
  }

  override fun getKey(): String {
    return path
  }

  companion object {
    @Serial
    private const val serialVersionUID: Long = -7333769143033322264L
  }
}
